presto-main/src/main/java/com/facebook/presto/operator/aggregation/differentialentropy/DifferentialEntropyStateFixedHistogramJacknifeStrategy.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.differentialentropy;
+
+import com.facebook.presto.operator.aggregation.fixedhistogram.FixedDoubleBreakdownHistogram;
+import com.google.common.collect.Streams;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+Calculates sample entropy using jacknife estimates on a fixed histogram.
+See http://cs.brown.edu/~pvaliant/unseen_nips.pdf.
+ */
+public class DifferentialEntropyStateFixedHistogramJacknifeStrategy
+        extends DifferentialEntropyStateFixedHistogramStrategy
+{
+    public DifferentialEntropyStateFixedHistogramJacknifeStrategy(long bucketCount, double min, double max)
+    {
+        super(new FixedDoubleBreakdownHistogram((int) bucketCount, min, max));
+    }
+
+    protected DifferentialEntropyStateFixedHistogramJacknifeStrategy(DifferentialEntropyStateFixedHistogramJacknifeStrategy other)
+    {
+        super(new FixedDoubleBreakdownHistogram(other.getBreakdownHistogram()));
+    }
+
+    public DifferentialEntropyStateFixedHistogramJacknifeStrategy(SliceInput input)
+    {
+        super(new FixedDoubleBreakdownHistogram(input));
+    }
+
+    @Override
+    public void validateParams(
+            long bucketCount,
+            double sample,
+            Double weight,
+            Double min,
+            Double max)
+    {
+        super.validateParams(bucketCount, sample, weight, min, max);
+    }
+
+    @Override
+    public void mergeWith(DifferentialEntropyStateStrategy other)
+    {
+        getBreakdownHistogram()
+                .mergeWith(((DifferentialEntropyStateFixedHistogramJacknifeStrategy) other).getBreakdownHistogram());
+    }
+
+    @Override
+    public void add(double value, double weight)
+    {
+        getBreakdownHistogram().add(value, weight);
+    }
+
+    @Override
+    public double calculateEntropy()
+    {
+        final Map<Double, Double> bucketWeights = new HashMap<Double, Double>();
+        Streams.stream(getBreakdownHistogram().iterator()).forEach(
+                e -> {
```

Comment:
Isn't this a `bucket`?

Reply:
It's an entry mapping a value to a histogram of weights.